### ISSUE TITLE
How to start ZMQ as a dedicated task in Tokio?

### ISSUE THREAD

**Issue by marvin-hansen** [zeromq/zmq.rs#184]
Hi, 

I've previously used ZMQ via TMQ but because TMQ doesn't compile statically with MUSEL, I'm switching to this crate. This crate compiles fine with MUSEL, so that's good. However, I face a weird issue when trying to start ZMQ outside a simple main test file. 

I've a fairly straightforward use case:

A tokio server provides both, an http endpoint and the ZMQ message handler. For practical reasons, the ZMQ message handler is stuffed into a seperae server struct, constructed and then spawned as a separate task. This worked as expected with TMQ. 

With this crate, though, when I call 

    let mut sink = zeromq::PushSocket::new();
    sink.connect("tcp://127.0.0.1:5558").await.unwrap();

Connect blocks forever and I would have to remove the await to unblock, but then it doesn't connect. This really puzzles me because NATS, TMQ, and any other message library I've uses never blocked on connect, but this one does and I wonder how can I fix this?

Any help would be appreciated. 

Thank you.
